R/player_transfer_history.R

```diff
@@ -9,155 +21,183 @@
-
-        market_value <- tryCatch(all_transfer_rows[each_row] %>% rvest::html_nodes(".tm-player-transfer-history-grid__market-value") %>% rvest::html_text() %>%
-                                   stringr::str_squish() %>%
-                                   .convert_value_to_numeric(), error = function(e) NA_character_)
-
-        transfer_value_info <- tryCatch(all_transfer_rows[each_row] %>% rvest::html_nodes(".tm-player-transfer-history-grid__fee") %>% rvest::html_text() %>%
-                                     stringr::str_squish() %>% tolower, error = function(e) NA_character_)
-
-        transfer_value <- transfer_value_info %>% .convert_value_to_numeric
-
-        if(grepl("free", transfer_value_info)) {
-          transfer_type <- 'Free transfer'
-        } else if(grepl("end of", transfer_value_info)) {
-          transfer_type <- 'End of loan'
-        } else if(grepl("fee", transfer_value_info)) {
-          transfer_type <- 'Paid loan'
-        } else if(grepl("loan", transfer_value_info)) {
-          transfer_type <- 'Loan'
-        } else if(grepl("m", transfer_value_info)) {
-          transfer_type <- 'Transfer'
-        } else if(grepl("th.", transfer_value_info)) {
-          transfer_type <- 'Transfer'
-        } else if(grepl("k", transfer_value_info)) {
-          transfer_type <- 'Transfer'
-        } else {
-          transfer_type <- NA_character_
-        }
-
-        # Executed if the user wants to get more info. Contains: Contract Expiring date + Days remaining. From which countries was the transfer.
-        if(get_extra_info == TRUE){
+    player_name <- page %>%
+      rvest::html_node("h1") %>%
+      rvest::html_text() %>%
+      gsub("#[[:digit:]]+ ", "", .) %>%
+      stringr::str_squish()
+
+    player_id <- basename(player_url)
+    transfer_history_resp <- httr::GET(
+      url = paste0("https://www.transfermarkt.com/ceapi/transferHistory/list/", player_id),
+      httr::add_headers(.headers = c(
+        `User-Agent` = getOption("worldfootballR.agent")
+      ))
+    )
+
+    transfer_history_content <- content(transfer_history_resp)
+    raw_transfer_history <- transfer_history_content[["transfers"]]
+
+    formatted_transfer_history <- purrr::map(
+      raw_transfer_history,
+      \(.x) {
+        list(
+          "url" = .x[["url"]],
+          "season" = .x[["season"]],
+          "transfer_date" = lubridate::ymd(.x[["dateUnformatted"]]),
+          "team_from" = .x[["from"]][["clubName"]],
+          "team_to" = .x[["to"]][["clubName"]],
+          "market_value" = .convert_value_to_numeric(.x[["marketValue"]]),
+          "transfer_value" = .convert_value_to_numeric(.x[["fee"]]),
+          "transfer_type" = .standardize_fee(.x[["fee"]])
+        )
+      }
+    )
+
+    # Executed if the user wants to get more info. Contains: Contract Expiring date + Days remaining. From which countries was the transfer.
+    res <- if (isTRUE(get_extra_info)) {
```

Comment:
no actual logic changes for `get_extra_info` retrieval. just better spacing, etc.